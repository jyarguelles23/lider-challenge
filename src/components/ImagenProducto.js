import React from "react";
import Price from "../components/Price";
import OldPrice from "../components/OldPrice";

const Imagen = ({ producto }) => {
  const { id, brand, description, price, image, oldPrice } = producto;
  return (
    <div className="col-12 col-sm-6 col-md-4 col-lg-3 mb-4">
      <div className="card">
        <img src={image} alt={brand} className="card-img-top" />
        <div className="card-body">
          <p className="card-text">
            <strong >{brand}</strong> {description}{" "}
          </p>
          <Price key={`${id}-${price}`} price={price} oldPrice={oldPrice}></Price>
          <OldPrice key={`${id}-${oldPrice}`} oldPrice={oldPrice}></OldPrice>
         
        </div>
      </div>
    </div>
  );
};

export default Imagen;
