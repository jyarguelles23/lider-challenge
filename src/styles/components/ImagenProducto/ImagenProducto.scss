.product-price{
    font-size: $font-size-20;
}

.product-old-price{
    text-decoration: line-through;
      font-size: $font-size-18;
        padding: $padding-sm 0 $padding-lg 0;
}

.image-porciento{
  max-height: 680px;
  width: 30%;
  height: 30%;
  border-radius: 4px;
}

.card-text{
  font-size: $font-size-16;
}