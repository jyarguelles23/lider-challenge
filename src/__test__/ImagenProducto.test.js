import ImagenProducto from "../components/ImagenProducto";
import React from "react";
import renderer, { ReactTestRenderer } from "react-test-renderer";

let tree = ReactTestRenderer;

const producto = {
  id: "5f63bf9c55b47b07c4fb83b8",
  brand: "ooy eqrceli",
  description: "rl√±lw brhrka",
  image: "https://lider.cl/catalogo/images/whiteLineIcon.svg",
  price: 498.724,
  oldPrice: 997.448,
};

describe("<ImagenProducto/>", () => {
  beforeEach(() => {
    tree = renderer.create(<ImagenProducto producto={producto} />);
  });

  it("should render without crashing", () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should render OldPrice tag", () => {
    tree = renderer.create(<ImagenProducto producto={producto} />);
    const oldPrice = tree.root;
    expect(
      oldPrice.findByProps({ className: "product-old-price" }).children
    ).toEqual([`$ ${producto.oldPrice}`]);
  });


});
