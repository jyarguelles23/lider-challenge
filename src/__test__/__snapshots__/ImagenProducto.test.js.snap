// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImagenProducto/> should render without crashing 1`] = `
<div
  className="col-12 col-sm-6 col-md-4 col-lg-3 mb-4"
>
  <div
    className="card"
  >
    <img
      alt="ooy eqrceli"
      className="card-img-top"
      src="https://lider.cl/catalogo/images/whiteLineIcon.svg"
    />
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        <strong>
          ooy eqrceli
        </strong>
         
        rl√±lw brhrka
         
      </p>
      <div
        className="product-price"
      >
        $ 
        498.724
        <img
          alt="50%"
          className="image-porciento"
          src="50porciento.png"
        />
      </div>
      <div
        className="product-old-price"
      >
        $ 997.448
      </div>
    </div>
  </div>
</div>
`;
